vars/helm.groovy

```diff
@@ -31,8 +31,8 @@ def deployChart(environment, registry, chartName, tag) {
 }
 
 // private
-def writeUrlIfIngress(deploymentName) {  
-  sh "if kubectl get ingress $deploymentName --ignore-not-found --namespace $deploymentName; then echo 'Build available for review at https://$deploymentName.$INGRESS_SERVER'; fi"
+def writeUrlIfIngress(deploymentName) {
+  sh "kubectl get ingress -n $deploymentName -o jsonpath='{.items[0].spec.rules[0].host}' --ignore-not-found | sed -e 's/^/Build available for review at https:\\/\\//;'"
```

Comment:
What is the rationale for using `jsonpath` as opposed to `jq`? I ask as there are a handful of uses of `jq` already within the library and I thought it might be better to stick with it for consistency?

I don't know if there are benefits to using `sed` rather than something like `printf` in this situation but if there aren't something like this does the same thing:
`kubectl get ingress -n $deploymentName -o json --ignore-not-found | jq '.items[0].spec.rules[0].host' | xargs printf "Build available for review at https://%s\n"`.

Maybe something to consider?

Reply:
That approach doesn't quite work as you'll end up printing `Build available for review at https://` if no ingress exists.

I'm not opposed to `jq` or `printf` this was just an approach I could get to work.

Reply:
Ah yeah, I was testing against a nonexistent deployment rather than a deployment with no ingress - they are handled differently.
This should work: `kubectl get ingress -n $deploymentName -o json --ignore-not-found | jq '.items[0].spec.rules[0].host // empty' | xargs printf "Build available for review at https://%s\n"`.
I've added a default `empty` using the [alternative operator](https://stedolan.github.io/jq/manual/#Alternativeoperator://)

Reply:
It still returns the same value for a non-existent ingress.

![image](https://user-images.githubusercontent.com/42574977/79074521-c94ee500-7ce4-11ea-8af6-a46c87fde22b.png)

Reply:
Mmh, I wonder if there is a difference in the implementation of `xargs` between our systems :(

Reply:
![Screenshot 2020-04-12 at 17 48 54](https://user-images.githubusercontent.com/639706/79074743-01a2f300-7ce6-11ea-9690-51a1ec1409f6.png)

Reply:
Apparently, no the subject of `xargs`
> It’s default behavior is slightly different in Mac (or BSD) and Linux, in a subtle way. On the Mac, if there is no input from stdin, it will not execute the command. On Linux, it will execute it without any argument.

[source](https://ylan.segal-family.com/blog/2016/10/19/subtleties-of-xargs-on-mac-and-linux/)

That explains the difference. Looks like it can be fixed with the flag ` --no-run-if-empty`. Resulting in the command `kubectl get ingress -n $deploymentName -o json --ignore-not-found | jq '.items[0].spec.rules[0].host // empty' | xargs --no-run-if-empty printf "Build available for review at https://%s\n"`

Reply:
Aha!  That works locally.  I'll add it to the pipeline and test it out.

Reply:
That worked perfect!